pkg/provisioner/ironic/ironic.go

```diff
@@ -1305,10 +1313,9 @@ func (p *ironicProvisioner) setMaintenanceFlag(ironicNode *nodes.Node, value boo
 	case gophercloud.ErrDefault409:
 		p.log.Info("could not set host maintenance flag, busy")
```

Comment:
In this case nothing has been done and we need to retry after a delay, so we can't just return nil.